CMakeLists.txt

```diff
@@ -1,88 +1,218 @@
+kompute_option(KOMPUTE_OPT_USE_BUILD_IN_PYBIND11 "Use the build in version of pybind11." ON)
+kompute_option(KOMPUTE_OPT_USE_BUILD_IN_VULKAN_HEADER "Use the build in version of Vulkan Headers. This could be helpful in case your system Vulkan Headers are to new for your driver. If you set this to false, please make sure your system Vulkan Header are supported by your driver." ON)
+kompute_option_string(KOMPUTE_OPT_BUILD_IN_VULKAN_HEADER_TAG "The git tag used for the build in Vulkan Headers when 'KOMPUTE_OPT_USE_BUILD_IN_VULKAN_HEADER' is enabled. A list of tags can be found here: https://github.com/KhronosGroup/Vulkan-Headers/tags" "v1.2.203")
+message(STATUS "=======================================================")
+
+#####################################################
+# Deprecated Options
+#####################################################
+include(cmake/deprecation_warnings.cmake)
+
+#####################################################
+# Dependencies
+#####################################################
+include(cmake/vulkan_shader_compiler.cmake)
+include(FetchContent)
+include(cmake/check_vulkan_version.cmake)
+
+# Vulkan Header
+# We don't import Vulkan library if Android build as its build dynamically
+# Otherwise it is expected that the Vulkan SDK and dependencies are installed
+if(NOT KOMPUTE_OPT_ANDROID_BUILD)
+    find_package(Vulkan REQUIRED)
+endif()
+if(KOMPUTE_OPT_USE_BUILD_IN_VULKAN_HEADER)
+    FetchContent_Declare(vulkan_header GIT_REPOSITORY https://github.com/KhronosGroup/Vulkan-Headers.git
+                                       GIT_TAG ${KOMPUTE_OPT_BUILD_IN_VULKAN_HEADER_TAG}) # Source: https://github.com/KhronosGroup/Vulkan-Headers/tags
+    FetchContent_MakeAvailable(vulkan_header)
+    
+    if(NOT KOMPUTE_OPT_DISABLE_VULKAN_VERSION_CHECK)
+        # Ensure the driver supports this Vulkan version
+        check_vulkan_version(INCLUDE_DIR "${vulkan_header_SOURCE_DIR}/include")
+    endif()
+else()
+    if(NOT KOMPUTE_OPT_DISABLE_VULKAN_VERSION_CHECK)
+        # Ensure the driver supports this Vulkan version
+        check_vulkan_version(INCLUDE_DIR ${Vulkan_INCLUDE_DIR})
     endif()
 endif()
 
-if(KOMPUTE_OPT_INSTALL)
-	# Enable install parameters for fmt (overrides parameters passed)
-	set(FMT_INSTALL ON CACHE BOOL "Enables install of fmt" FORCE)
+# Spdlog
+if(KOMPUTE_OPT_USE_BUILD_IN_SPDLOG)
+    set(SPDLOG_INSTALL ${KOMPUTE_OPT_INSTALL})
+    set(SPDLOG_BUILD_SHARED ${KOMPUTE_OPT_DEPENDENCIES_SHARED_LIBS})
+
+    FetchContent_Declare(spdlog GIT_REPOSITORY https://github.com/gabime/spdlog.git
+                                GIT_TAG v1.10.0) # Source: https://github.com/gabime/spdlog/releases
+    FetchContent_MakeAvailable(spdlog)
+else()
+    find_package(spdlog REQUIRED)
+endif()
+
+# fmt
+if(KOMPUTE_OPT_USE_BUILD_IN_FMT)
+    set(FMT_INSTALL ${KOMPUTE_OPT_INSTALL})
+    set(BUILD_SHARED_LIBS_BKP ${KOMPUTE_OPT_DEPENDENCIES_SHARED_LIBS})
```

Comment:
What is the _BKP postfix? Is there a reason for this one not to also be a KOMPUTE_ var?